### DISCUSSION TITLE
GLB artifacts that only show up in model-viewer?

### DISCUSSION THREAD

**Discussion by christaggart** [google/model-viewer#3815] [Q&A]
We're seeing strange artifacting when models are exported from BabylonJS to GLB and displayed with `<modelviewer>` that don't show up in USDZ converted models (GLB -> USDZ). It seems like these are evident when placing faces close together? In this case we need those faces to touch, and that seems to cause certain artifacting. These do, however, look fine inside the BabylonJS Playground and Gestaltor.. Is this something we're doing wrong when creating the model, or maybe a bug in modeviewer?

https://cp-development.s3.amazonaws.com/Screen%20Recording%202022-09-20%20at%2011.23.01%20AM.mov

Here's an example you can use the debugger to export to GLB
https://www.babylonjs-playground.com/#2CU0P1#92

Here's an exported file loaded into the mode-viewer: https://cp-development.s3.amazonaws.com/artifacts.html

**Comment by elalish** [comment]
Yes, this is called Z-fighting, and is a very common rendering artifact, which is very sensitive to camera parameters like near/far plane, depth buffer resolution, etc. It's likely just luck that you're seeing it in one renderer and not another. Basically it's never a good idea to have surfaces that are almost coplanar; easiest solution is just to clip out those internal faces since they're invisible anyway. It'll even improve performance slightly.
